src/sdltiles.cpp

```diff
@@ -2296,6 +2296,39 @@ void update_music_volume() {
 }
 
 #ifdef SDL_SOUND
+std::unordered_map<std::string, Mix_Chunk*> unique_chunks;
+
+// Allocate new Mix_Chunk copy of input, sets ::allocated to 0 so it is not freed
+// during Mix_FreeChunk at EOL
+static Mix_Chunk* copy_chunk(const Mix_Chunk* ref){
+    // SDL_malloc to match up with Mix_FreeChunk's SDL_free call
+    Mix_Chunk *nchunk = (Mix_Chunk*)SDL_malloc(sizeof(Mix_Chunk));
+
+    (*nchunk) = *ref;
+    nchunk->allocated = 0;
+    return nchunk;
```

Comment:
I think I'm confused here.  You're doing an SDL_malloc to match up with the SDL_free, which sounds right, but then setting `nchunk->allocated` to 0 so it doesn't free it?  I think one of these is wrong.

Now I think about it, doing the SDL_malloc still isn't saving any memory allocs, how is this saving memory?  In your test instrumentation your only checking if `allocated` is null which you set it to.  This isn't actually profiling memory, I would like to see some better instrumentation from a tool that was built to do this like valgrind.

Reply:
Ah got it so the SDL_malloc is getting memory for the struct but the allocated field is telling it the abuf is already allocated by someone else.  I understand now.  Nevermind I think this is probably good.  Some valgrind data to backup your instrumentation would still be nice.

Reply:
Instrumentation was testing against `allocated` so that it didn't count the same memory region multiple times. I should have explained better, either in the PR description or the comments, how `Mix_FreeChunk` works wrt to `allocated`.